builder/bundles.go

```diff
@@ -671,17 +650,33 @@ func installBundleToFull(packagerCmd []string, baseDir string, bundle *bundle, d
 	}
 
 	// feed the channel
-	for rpm := range bundle.AllRpmPackages {
+	for rpm, pkgInfo := range bundle.AllRpms {
+		var rpmFull string
+
 		if rpmMap[rpm] {
 			continue
 		}
-		rpmFull := filepath.Join(localPath, rpm)
+
+		if repos[pkgInfo.repo].urlScheme != "file" {
+			rpmFull = filepath.Join(dnfDownloadDir, rpm)
+		} else {
+			rpmFull = filepath.Join(repos[pkgInfo.repo].url, rpm)
+		}
+
 		if _, err = os.Stat(rpmFull); os.IsNotExist(err) {
-			rpmFull = filepath.Join(rpmPath, rpm)
+			// if rpm is not found, name of rpm file is not in autospec generated format. In this case
+			// we query the database to get the name of the rpm file containing the package.
+			rpm = queryRpmName(packagerCmd, pkgInfo.name, pkgInfo.repo)
```

Comment:
When the repo has multiple files with the same name, `dnf repoquery` will resolve the paths to multiple rpms. We can be more specific by using the NVRA (`pkg.name+"-"+pkg.version+"."+pkg.arch`) to get a single result.